pkg/app/pipedv1/plugin/kubernetes/deployment/plugin.go

```diff
@@ -27,47 +27,29 @@ import (
 	"github.com/pipe-cd/pipecd/pkg/plugin/sdk"
 )
 
-// Plugin implements the sdk.DeploymentPlugin interface.
-type Plugin struct {
+// DeploymentPlugin implements the sdk.DeploymentPlugin interface without Name and Version.
+type DeploymentPlugin struct {
 }
 
 type loader interface {
 	// LoadManifests renders and loads all manifests for application.
 	LoadManifests(ctx context.Context, input provider.LoaderInput) ([]provider.Manifest, error)
 }
 
-type toolRegistry interface {
-	Kubectl(ctx context.Context, version string) (string, error)
-	Kustomize(ctx context.Context, version string) (string, error)
-	Helm(ctx context.Context, version string) (string, error)
-}
-
-var _ sdk.DeploymentPlugin[sdk.ConfigNone, kubeconfig.KubernetesDeployTargetConfig] = (*Plugin)(nil)
```

Comment:
The type `DeploymentPlugin` in this package no longer implements `sdk.DeploymentPlugin` because `sdk.DeploymentPlugin` requires `Name` and `Version` methods. So I removed this.

Reply:
[IMO] If possible, it would be nice to keep it as is to clarify what should be implemented.  Maybe we need to consider the the interface and SDK.

Reply:
I noticed there is a better SDK interface for registering a plugin.
I'll try to change the SDK.

I imagine something like the one below.
```go
sdk.RegisterPlugin("kubernetes", "0.0.1", sdk.WithDeploymentPlugin(&deployment.Plugin{}), sdk.WithLivestatePlugin(&livestate.Plugin{}))
```

Reply:
Thank you! it is worth trying 👍

Reply:
I implemented this in another PR, #5698.
After merging it, I'll rewrite this PR.

Reply:
I thought sending another PR to use the new SDK interface would be better, so I closed this.
The new implementation is #5701